src/main/java/com/app/library/controller/AuthorController.java

```diff
@@ -0,0 +1,71 @@
+package com.app.library.controller;
+
+import com.app.library.model.dto.AuthorDto;
+import com.app.library.service.AuthorService;
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@Validated
+@RequestMapping("/api/authors")
+@AllArgsConstructor
+public class AuthorController {
+
+    private final AuthorService authorService;
+
+    @GetMapping
+    ResponseEntity<List<AuthorDto>> getAllAuthors() {
+        List<AuthorDto> allAuthors = authorService.getAllAuthors();
+        if (!allAuthors.isEmpty()) {
+            return ResponseEntity.ok(allAuthors);
+        }
+        return ResponseEntity.noContent().build();
+    }
+
+    @GetMapping("/{id}")
+    ResponseEntity<AuthorDto> getAuthorById(@PathVariable UUID id) {
+        return authorService.getAuthorById(id)
+                .map(ResponseEntity::ok)
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+    @PostMapping
+    ResponseEntity<AuthorDto> saveAuthor(@Valid @RequestBody AuthorDto author) {
+        AuthorDto savedAuthor = authorService.saveAuthor(author);
```

Comment:
I think instead of that logic, you can simply use `return new ResponseEntity<>(T body, HttpStatusCode status);` :) Try to use it to those methods where it can be used :) - in all controller classes

Reply:
Done